src/platforms/hosted/cli.c

```diff
@@ -133,76 +134,102 @@ static void cl_help(char **argv)
-		"\t                  FTDI chip connected through resistors to TMS, TDI and TDO\n"
-		"\t-p              Power the target from the probe (if possible)\n"
-		"\t-R[h]           Reset the device. If followed by 'h', this will be done using\n"
-		"\t                  the hardware reset line instead of over the debug link\n"
-		"\t-H              Do not use the high level command API (bmp-remote)\n"
-		"\t-M <string>     Run target-specific monitor commands. This option\n"
-		"\t                  can be repeated for as many commands you wish to run.\n"
-		"\t                  If the command contains spaces, use quotes around the\n"
-		"\t                  complete command\n"
+		"General configuration options:\n"
+		"\t-n, --number     <scan-number> Select the target device at the given position in the\n"
+		"\t                   scan chain (use the -t option to get a scan chain listing)\n"
+		"\t-j, --jtag       Use JTAG instead of SWD\n"
+		"\t-C, --hw-reset   Connect to target under hardware reset\n"
+		"\t-t, --list-chain Perform a chain scan and display information about the\n"
+		"\t                   conected devices\n"
+		"\t-T, --timing     Perform continues read- or write-back of a value to allow\n"
+		"\t                   measurement of protocol timing. Aborted by ^C\n"
+		"\t-e, --ext-res    Assume external resistors for FTDI devices, that is having the\n"
+		"\t                   FTDI chip connected through resistors to TMS, TDI and TDO\n"
+		"\t-p, --power      Power the target from the probe (if possible)\n"
+		"\t-R, --reset      [h] Reset the device. If followed by 'h', this will be done using\n"
+		"\t                   the hardware reset line instead of over the debug link\n"
+		"\t-H, --high-level Do not use the high level command API (bmp-remote)\n"
+		"\t-M, --monitor    <string> Run target-specific monitor commands. This option\n"
+		"\t                   can be repeated for as many commands you wish to run.\n"
+		"\t                   If the command contains spaces, use quotes around the\n"
+		"\t                   complete command\n"
 		"\n"
-		"SWD-specific configuration options [-f FREQUENCY | -m TARGET]:\n"
-		"\t-f <frequency>  Set an operating frequency for SWD\n"
-		"\t-m <target>     Use the given target ID for selection in SWD multi-drop\n"
+		"SWD-specific configuration options:\n"
+		"\t-f, --freq       <frequency> Set an operating frequency for SWD\n"
+		"\t-m, --mult-drop  <target> Use the given target ID for selection in SWD multi-drop\n"
 		"\n"
-		"Flash operation selection options [-E | -w | -V | -r]:\n"
-		"\t-E              Erase the target device Flash\n"
-		"\t-w              Write the specified binary file to the target device\n"
-		"\t                  Flash (the default)\n"
-		"\t-V              Verify the target device Flash against the specified\n"
-		"\t                  binary file\n"
-		"\t-r              Read the target device Flash\n"
+		"Flash operation options:\n"
```

Comment:
We'd included the word 'selection' here because these options select which operation to perform while the ones below modify that operation - might be worth figuring out how to clarify that.